blog/content/post/2020-08-05-PCIe-features.md

```diff
@@ -0,0 +1,213 @@
+---
+title: What features PCIe has?
+abstract: 'Introductory blog post to PCIe features.
+          In this article you can read what PCIe capability is
+          and see examples of such capabilities.'
+cover: /covers/PCIExpress.jpg
+author: marek.kasiewicz
+layout: post
+published: true
+date: 2020-08-05
+archives: "2020"
+
+tags:
+  - Firmware
+categories:
+  - Firmware
+
+---
+
+## What this blog is about
+
+Goal of this post is to introduce readers to the concept of PCI Express
+capabilities. It briefly describes what capability is and how it works. It also
+lists some example capability described in PCI Express (PCIe) specification.
+
+## Introduction
+
+PCI Express uses inverted tree topology. It starts from the Root Complex which
+is connected directly to the CPU and RAM. Root Complex branches to Switches and
+Endpoints. Switches branch founder and can be connected to more Switches or to
+Endpoints. Endpoints reside at the bottom of branches and have only one
+upstream Port (facing toward the Root). By comparison, a Switch may have several
+downstream Ports, but can only one upstream Port. Possible are also Bridges
+and Legacy Endpoints. Bridges are interfaces to other buses such as PCI. Legacy
+Endpoints are devices which were designed for the operation of an older bus like
+PCI‐X but now have a PCIe interface.
+
+![PCIe topology](/img/Example_PCI_Express_Topology.png)  
+Every Switch, Bridge and Endpoint is a PCIe Device. All Devices have at least one
+Function, and each Function has its Configuration Space which is Device registers
+mapped to either I/O memory space (used mostly by PCI Devices) or to normal
+memory space. By writing or reading from this memory OS can exchange information
+with a Function. For example OS can check the vendor of a Device by checking the
```

Comment:
check checking  ( try to use [thesaurus](thesaurus.com)) it will ease your writing